data/jakarta/utils.sh

```diff
@@ -126,3 +126,50 @@ snap_remove()
     snap remove edgexfoundry || true
 }
 
+# wait for services to come online
+# NOTE: this may have to be significantly increased on arm64 or low RAM platforms
+# to accomodate time for everything to come online
+snap_wait_all_services_online()
+{
+    while 
+        [ "$(curl --insecure --silent --include \
+            --output /dev/null --write-out "%{http_code}" \
+            -X GET 'http://localhost:59880/api/v2/ping' \
+            -H 'accept: application/json')" != 200 ] \
+        || [ "$(curl --insecure --silent --include \
+            --output /dev/null --write-out "%{http_code}" \
+            -X GET 'http://localhost:59881/api/v2/ping' \
+            -H 'accept: application/json')" != 200 ] \
+        || [ "$(curl --insecure --silent --include \
+            --output /dev/null --write-out "%{http_code}" \
+            -X GET 'http://localhost:59882/api/v2/ping' \
+            -H 'accept: application/json')" != 200 ] \
```

Comment:
Consider the following two error cases as well:
1. One of those services return an error status 4xx/5xx: Such errors should probably fail the test immediately.
2. One of those services don't accept the connection quickly: use curl's connect timeout  
3. One of those services accepts the connection but doesn't respond (hung internally): set curl max time

See https://unix.stackexchange.com/a/94612/194522

Reply:
Also, put a global max retry because the the while loop can go on forever if one of the conditions are not met for an unexpected reason. I'd say 5min is sufficient to wait before giving up. That could be a unix time comparison condition (current-start>300) or a counter counting to 300 (since the sleep is 1s).

Reply:
Thanks for your review @farshidtz I have updated them now.
I have set `--connect-timeout`=2 and `--max-time`=300. Does this seem reasonable?

Reply:
if one of those services take forever to respond, it would take 300 iterations of 300s to stop. So, it is too long. I think the service can be considered as hung if it takes more than few seconds to respond to the ping request. I suggest setting it to 5 max.